### PR TITLE
增加腾讯QQ/微信/微博客户端

### PR DESCRIPTION
**Pull request by xutongle**
| Q             | A
| ------------- | ---
| New feature?  | yes
| Tests pass?   | yes

### CODE REVIEW

**Comment by samdark** [comment]
Thanks for the effort but we aren't going to include more providers into the extension. I suggest releasing it as a separate package. After it's done I can add a link to that package to docs. Thanks!

**Comment by xutongle** [comment]
这几个接口，中国用户使用率100%，由于GFW的问题，多个包安装是非常慢的。

**Comment by samdark** [comment]
OK. We can make an exception then...

**Comment by xutongle** [comment]
thx

**Comment by samdark** [comment]
Would be great if you'll add changelog. I'll merge right after.

**Comment by xutongle** [comment]
ok

**Comment by xutongle** [comment]
Add change log to complete.

**Comment by samdark** [comment]
Merged with some adjustments. Thanks!

**Comment by yujiandong** [comment]
Hello, I am the author of qq/weibo/wechat auth-client.

https://github.com/yujiandong/yii2-authclient

**Comment by raincious** [comment]
@yujiandong 
Do you mean you are the another of these codes instead of OP?

If it's true, seems some further attention is needed due to copyright violation.

@samdark

**Comment by yujiandong** [comment]
@raincious @samdark 
 Yes, you can compare the codes https://github.com/yujiandong/yii2-authclient
I wrote it for my simpleforum 7 months ago.

But @xutongle change the author name in the file.

File: clients/Weibo.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+ *         ],
+ *     ]
+ *     ...
+ * ]
+ * ~~~
+ *
+ * @see http://open.weibo.com/
+ * @see http://open.weibo.com/wiki/
+ *
+ * @author Xu Tongle <xutongle@gmail.com>
```
Comment (CodeFalling) [inline clients/Weibo.php]: You are not the author 😒

**Comment by raincious** [comment]
@yujiandong 

LOL. You are not the only victim, see the 'Leaps Framework' this guy contributed to:
https://github.com/leaps/leaps/blob/3be68042b7060d2a6e4aef054efae812e57fda95/framework/Kernel.php

Looks just like:
https://github.com/yiisoft/yii2-framework/blob/master/BaseYii.php

Isn't?

Relax, there are a lot of people who take a $300/m full time job to write PHP code. For them, change the author declaration in other people's code while keep it's RUNability is hard enough for them to believe they can "own" the code. 

That's why they by the way changed the copyright declaration and remove you from your code as well.

Just want for the assignee make things right.

**Comment by lambdaq** [comment]
This @xutongle  guy has this badass commit

https://github.com/xutongle/yii2-authclient/commit/61528adec3f0d9ecc8ff1fa0215e75ec79869649

The commit message in Chinese reads "solving licensing problems".

I mean srsly? WTF?

File: clients/Weibo.php (hunk 1 above)
Comment (lambdaq) [inline clients/Weibo.php]: @CodeFalling  https://github.com/xutongle/yii2-authclient/commit/61528adec3f0d9ecc8ff1fa0215e75ec79869649

**Comment by cebe** [comment]
Thanks for bringing this to our attention, I have removed the commit from master for now. Need some time to keep up with the comments.

**Comment by tooooolong** [comment]
@cebe 👍

**Pull request closed without merging**